capstone/src/main/java/com/google/sps/servlets/DeleteDriverServlet.java

```diff
@@ -34,21 +36,24 @@
 public class DeleteDriverServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Long driverId_temp = Long.parseLong(request.getParameter("driverId"));
-    Filter propertyFilter = new FilterPredicate("id", FilterOperator.EQUAL, driverId_temp);
-    Query query = new Query("Driver").setFilter(propertyFilter);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-    ArrayList<Driver> drivers = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      String first= (String) entity.getProperty("first");
-      String day = (String) entity.getProperty("day");
-      String times = (String) entity.getProperty("times");
-      Long seats = (Long)entity.getProperty("seats");
-      Long id = (Long)entity.getProperty("id");
-      Driver driver = new Driver(first,day,times,seats,id);
-      Key driverKey = entity.getKey();
-      datastore.delete(driverKey);
+    UserService userService = UserServiceFactory.getUserService();
```

Comment:
I would actually invert the logic a little bit here. Instead of putting everything inside if (userService.isUserLoggedIn()), I would do it like this:

```
User user = userService.getCurrentUser();
if (user == null) {
  response.sendError(HttpServletResponse.SC_UNAUTHORIZED, "not logged in");
  return;
}

// The rest follows.
```

This way the rest of the code doesn't have to be indented and you don't have to worry about dealing with sending an error response in an else branch.

Reply:
Let me know if the updated files look alright?